addons/misra.py

```diff
@@ -256,14 +256,105 @@ def getForLoopExpressions(forToken):
             lpar.astOperand2.astOperand2.astOperand2]
 
 
+def findCounterTokens(cond):
+    if cond.str in ['&&', '||']:
+        c = findCounterTokens(cond.astOperand1)
+        c.extend(findCounterTokens(cond.astOperand2))
+        return c
+    ret = []
+    if cond.isArithmeticalOp:
+        if cond.astOperand1.isName and cond.astOperand2.isNumber:
+            ret.append(cond.astOperand1)
+        elif cond.astOperand2.isName and cond.astOperand1.isNumber:
```

Comment:
Do you need all these conditions that looks if cond.astOperand1 is a name or not? why not just:

    if cond.astOperand1.isName:
        ret.append(cond.astOperand1)
    if cond.astOperand2.isName:
        ret.append(cond.astOperand2)
    ....